scripts/aws/monitor.py

```diff
@@ -0,0 +1,187 @@
+        Role=role,
+        Handler=handler,
+        Code={
+            'ZipFile': open(zip_file, 'rb').read(),
+        },
+        Description=description,
+        Timeout=timeout,
+        MemorySize=mem_size,
+        Publish=True)
+
+    logging.info('response for creating lambda function = "%s"',
+                 response)
+
+
+def get_topic_arn(client, topic_name):
+    """Gets the Arn of an SNS topic"""
+    response = client.list_topics()
+    for value in response['Topics']:
+        sns_arn = value['TopicArn']
+        if topic_name in str(sns_arn):
+            return str(sns_arn)
+        else:
+            return None
+
+
+if __name__ == '__main__':
+
+    logging.basicConfig(level=logging.INFO, format='[%(asctime)s] %(levelname)s %(message)s')
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--aws-profile", required=True)
+    parser.add_argument("--aws-region", default="us-east-1")
+    parser.add_argument("--api-gw", required=True)
+    parser.add_argument("--splunk-host", required=True)
+    parser.add_argument("--splunk-token", required=True)
+
+    args = parser.parse_args()
+    session = botocore.session.Session(profile=args.aws_profile)
+
+    sns_client = session.create_client('sns', args.aws_region)
+    cw = session.create_client('cloudwatch', args.aws_region)
+
+    create_api_alarm(cw, 'api_count', 'Count', 'ApiGateway',
+                     'Average', 'GreaterThanOrEqualToThreshold',
+                     'Average API count for a period of 5 min', 50, 300, 1,
+                     [{'Name': 'ApiName', 'Value': args.api_gw}, {'Name': 'Label', 'Value': args.api_gw}],
+                     get_topic_arn(sns_client, 'aws-non-critical-alert'))
+
+    create_api_alarm(cw, 'api_latency', 'Latency', 'ApiGateway', 'Average',
+                     'GreaterThanOrEqualToThreshold', 'Average API Latency for a period of 5 min',
+                     3, 300, 1, [{'Name': 'ApiName', 'Value': args.api_gw}, {'Name': 'Label', 'Value': args.api_gw}],
+                     get_topic_arn(sns_client, 'aws-non-critical-alert'))
+
+    create_api_alarm(cw, 'api_errors_4xx', '4XXError', 'ApiGateway', 'Average',
+                     'GreaterThanOrEqualToThreshold', 'Average 4XX errors for a period of 5 min',
+                     4, 300, 1, [{'Name': 'ApiName', 'Value': args.api_gw}, {'Name': 'Label', 'Value': args.api_gw}],
+                     get_topic_arn(sns_client, 'aws-non-critical-alert'))
+
+    create_api_alarm(cw, 'api_errors_5xx', '5XXError', 'ApiGateway', 'Average',
+                     'GreaterThanOrEqualToThreshold', 'Average 5XX errors for a period of 5 min',
+                     4, 300, 1, [{'Name': 'ApiName', 'Value': args.api_gw}, {'Name': 'Label', 'Value': args.api_gw}],
+                     get_topic_arn(sns_client, 'aws-non-critical-alert'))
+
+    iam_client = session.create_client('iam', args.aws_region)
+    role_arn = create_role(iam_client, 'lambda_basic_execution_monitor',
```

Comment:
The function checks if the role already exists. Also changed the name of the role to bit more detailed. See if its okay.